### PR TITLE
Bench22

### PR DESCRIPTION
**Pull request by ulrikeyang**
This PR contains various changes that are needed to compute the figure of merit for a new AMG benchmark.

### CODE REVIEW

File: src/parcsr_ls/par_amg_setup.c [hunk 1]
```diff
@@ -3698,6 +3700,18 @@ hypre_BoomerAMGSetup( void               *amg_vdata,
       hypre_CreateLambda(amg_data);
    }
 
+   if (cum_nnz_AP > -1)
```
Comment (liruipeng) [inline src/parcsr_ls/par_amg_setup.c]: A minor q.: Where is `cum_nnz_AP` set to `-1`? Why do we check `> -1`? Thanks!

**Review by liruipeng — APPROVED** [review]
Looks good to me. Thanks!

File: src/parcsr_ls/par_amg_setup.c (hunk 1 above)
Comment (ulrikeyang) [reply src/parcsr_ls/par_amg_setup.c]: Good point. Thanks for catching that.

**Review by waynemitchell — APPROVED** [review]
Looks good to me.

**Review by victorapm — APPROVED** [review]
Looks good, thanks!

**Review by rfalgout — APPROVED** [review]
Looks good.

**Comment by ulrikeyang** [comment]
I am just running the tests on my tux box and plan to merge this in when it's all clean. Let me know if any of you sees an issue with this.

**Pull request merged**